src/main/java/org/killbill/billing/plugin/bridge/api/BridgePaymentPluginApi.java

```diff
@@ -430,7 +441,8 @@ private PaymentTransactionInfoPlugin internalPaymentTransactionOperation(final T
         };
 
         try {
-            final PaymentTransactionInfoPlugin result = internalGenericPaymentTransactionOperation(op, new PaymentTransactionInfoPluginResultConverter(payment), context.getTenantId());
+            final PaymentTransactionInfoPluginResultConverter converter = new PaymentTransactionInfoPluginResultConverter(payment);
+            final PaymentTransactionInfoPlugin result = internalGenericPaymentTransactionOperation(op, converter, context.getTenantId(), converter.convertModelToApi(new org.killbill.billing.client.model.PaymentTransaction()));
```

Comment:
I'm a bit unfamiliar with the code: will the `converter` populate at least the `getKbTransactionPaymentId` field if the `PaymentTransaction` is mostly blank? This is required to avoid confusing the Janitor (https://docs.killbill.io/latest/payment_plugin.html#_paymentpluginstatus_states_overview).

Reply:
This method is only called from `authorizePayment`, .. , i.e the transaction methods. So this would not be called from the Janitor. However you remark is still pertinent for the call `getPaymentInfo`, will verify..